ext/java/org/msgpack/jruby/Encoder.java

```diff
@@ -147,7 +149,10 @@ private void appendBignum(RubyBignum object) {
     BigInteger value = object.getBigIntegerValue();
     if (value.compareTo(RubyBignum.LONG_MIN) < 0 || value.compareTo(RubyBignum.LONG_MAX) > 0) {
       if (value.bitLength() > 64 || (value.bitLength() > 63 && value.signum() < 0)) {
-        throw runtime.newArgumentError(String.format("Cannot pack big integer: %s", value));
+        if (hasBigintExtType && tryAppendWithExtTypeLookup(object)) {
+          return;
+        }
+        throw runtime.newRangeError(String.format("Cannot pack big integer: %s", value));
```

Comment:
The MRI extension raises `RangeError` on bigint, but the JRuby one `ArgumentError`. I assume it is a bug.